php/Client.php

```diff
@@ -234,13 +219,23 @@ public function update(Record $record)
     }
 
     /**
-     * Deletes a record from the E3DB system
+     * Deletes a record from the E3DB system, with optional optimistic locking.
      *
      * @param string $record_id
+     * @param string $version If non-null, will be checked against the
+     *   version of the record stored on the server. The record will
+     *   only be deleted if the version matches the value given here.
+     *
+     * @throws ConflictException If the version given does not match
+     * the version stored on the server.
      */
-    public function delete(string $record_id)
+    public function delete(string $record_id, string $version=null)
     {
-        $path = $this->conn->uri('v1', 'storage', 'records', $record_id);
+        if($version != null)
+            $path = $this->conn->uri('v1', 'storage', 'records', 'safe', $record_id, $version);
+        else
```

Comment:
Should use explicit `{}` in if statements:

```php
if (...) {

} else {

}
```

PHP itself is fine without it, but we try to follow PSR-1/2 coding standards which require it.